src/main/java/org/aksw/simba/squirrel/metadata/CrawlingActivity.java

```diff
@@ -139,17 +220,41 @@ public Worker getWorker() {
         return worker;
     }
 
-    public enum CrawlingURIState {SUCCESSFUL, UNKNOWN, FAILED;}
+    public enum CrawlingURIState {SUCCESSFUL, UNKNOWN, FAILED}
 
     public CrawlingURIState getState() {
         return state;
     }
 
-    public CrawleableUri getCrawleableUri() {
-        return uri;
+    public URI getUrl (String o)
+    {
+            try {
+                URL Domain = new URL("http://www.example.org/");
+                URL url = new URL(Domain + o);
+
+                return getUri(url);
+
+            } catch (MalformedURLException e) {
+                 LOGGER.info("MalformedURLException" + e);
+                e.printStackTrace();
+                 return null;
+
+            }
     }
 
-    public String getGraphId() {
-        return graphId;
+    public URI getUri(URL u)
+    {
+       try
+       {
+           return URI.create(u.toString());
+       }
+       catch (IllegalArgumentException x)
+       {
+           LOGGER.error("IllegalArgumentException"+x);
+            x.printStackTrace();
```

Comment:
When logging an exception, it is better to use a different error method: 
```java
LOGGER.error("here comes your error description.", x);
```
This will log the exception and its stack trace which is much better than logging exception name and printing the stack trace to the error stream.